app/src/features/surveys/components/species/components/FocalSpeciesEcologicalUnitsForm.tsx

```diff
@@ -61,14 +59,19 @@ export const FocalSpeciesEcologicalUnitsForm = (props: ISelectedSpeciesProps) =>
       name={`species.focal_species.[${index}].ecological_units`}
       render={(arrayHelpers: FieldArrayRenderProps) => (
         <Stack gap={2}>
-          {selectedUnits.map((ecological_unit, ecologicalUnitIndex) => (
+          {selectedUnits.map((_, ecologicalUnitIndex) => (
             <EcologicalUnitsSelect
-              key={ecological_unit.critterbase_collection_category_id ?? v4()}
+              // Key is intentionally index
+              key={ecologicalUnitIndex}
```

Comment:
I couldn't find a solution that avoided index as the key but welcome one that can!

Reply:
Could memoize the selectedUnits to prevent it from recaculating on every render and generate a static _id property.

```js
const selectedUnits = useMemo(() => {
values.species.focal_species.filter((item) => item.tsn === species.tsn).flatMap((item) => ({_id: v4(), unit: item.ecological_units})) ??
    []
}, [values.species.focal_species])
```